### PR TITLE
Javascript javascript2 week1

### PR DESCRIPTION
**Pull request by yuvi20188**

### CODE REVIEW

File: javascript/javascript2/week1/home2week1.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// 1.1. Find the shortest word.
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const minLength = Math.min.apply(
+  Math,
+  danishWords.map(function (item) {
```
Comment (chezzoba) [inline javascript/javascript2/week1/home2week1.js:5]: Try arrow functions to make it shorter: `danishWords.map(item => item.length)`. Also, for loops are more suited for this task than the array methods you used (keep in mind this is last week's assignment). If you want to use array methods for this, [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is a great one: `danishWords.reduce((accum, curv) => curv.length < accum.length ? curv : accum)`

File: javascript/javascript2/week1/home2week1.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    return item.length;
+  })
+);
+const minWords = danishWords.filter(function (item) {
```
Comment (chezzoba) [inline javascript/javascript2/week1/home2week1.js:9]: The callback function within filter should return a boolean (true / false) value. Example: `danishWords.filter(item => item.length === minLength)`

File: javascript/javascript2/week1/home2week1.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  if (item.length === minLength) return item;
+});
+console.log(minWords.join(" "));
+
+// 1.2. Find and count the Danish letters.
+const danishString2 = "Blå grød med røde bær".match(/[åØÆÅøæ]/g).length;
```
Comment (chezzoba) [inline javascript/javascript2/week1/home2week1.js:15]: Nice and short solution! However, you forgot one small edge case where there are no danish letters, for which .match will return null instead. To solve this you can use this: `("Blå grød med røde bær".match(/[åøæ]/gi) || []).length` (gi = case insensitive search).

File: javascript/javascript2/week1/home2week1.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "a cunning serpent",
+  "a brave lion",
+  "an evil echidna",
+  "a cowardly hare",
+  "a fast horse",
+  "a sleeping sloth",
+];
+
+// get random aninal name from array.
+function randomAnimal() {
```
Comment (chezzoba) [inline javascript/javascript2/week1/home2week1.js:34]: Nice work, but there are helper functions for this in the hyfBayhelp.js file.

**Review by chezzoba — COMMENTED** [review]
Hi Iuliia, good use of Javascript. You have done most of it correctly. However, there are still a few edge cases you have failed to consider. Generally, I suggest you create functions for each task you are completing so that you may test your code on your own examples.

**Comment by yuvi20188** [comment]
Thank you very much

вт, 22 нояб. 2022 г., 14:56 Kaizad Wadia ***@***.***>:
